onadata/auth.py

```diff
@@ -33,19 +36,70 @@ def authenticate(self, request):
 class QedRemoteUserAttributeMiddleware(object):
     username_header = 'HTTP_X_AUTH_USERNAME'
     email_header = 'HTTP_X_AUTH_EMAIL'
+    roles_header = 'HTTP_X_AUTH_ROLES'
 
     def process_request(self, request):
         username = request.META.get(self.username_header, None)
         email = request.META.get(self.email_header, None)
+        roles_json = request.META.get(self.roles_header, None)
+
+        remote_user_ok = hasattr(request, 'user') and request.user.is_authenticated \
+            and username and email
+
+        if not remote_user_ok:
+            return
+
+        user = request.user
+
+        QedRemoteUserAttributeMiddleware.process_user(user, email)
+        QedRemoteUserAttributeMiddleware.process_roles(user, roles_json)
+
+    @staticmethod
+    def process_user(user, email):
+        needs_saving = False
+
+        if user.email != email:
+            user.email = email
+            needs_saving = True
+
+        if needs_saving:
+            user.save()
+
+    @staticmethod
+    def process_roles(user, roles_json):
+        # KT__${var.kt_form_id}__${var.kt_form_permission}
+        roles = json.loads(roles_json)
```

Comment:
in case of missing roles header in the requests, this will fail with TypeError/JSONDecodeError, should this be handled better?